src/base/Cellar.sol

```diff
@@ -288,8 +299,14 @@ contract Cellar is ERC4626, Ownable, Multicall {
         // Distrust position.
         isTrusted[position] = false;
 
-        // Remove position from the list of positions if it is present.
-        positions.remove(position);
+        // Only remove position if it is not being used, is empty, and if it is not the holding position.
+        if (isPositionUsed[position]) {
+            uint256 positionBalance = _balanceOf(position);
+            if (positionBalance > 0) revert Cellar__PositionNotEmpty(position, positionBalance);
+            if (position == holdingPosition) revert Cellar__RemoveHoldingPosition();
+            positions.remove(position);
```

Comment:
This will cause distrust to revert if the position being distrusted has a balance. I don't think this is what we want. It should still be distrusted, but just not removed from the `positions` array (otherwise it's balance will not be accounted for in `totalAssets`).